### PR TITLE
Remove usages of native bootstrapping

### PR DESCRIPTION
**Pull request by hoyosjs**

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

File: src/mono/mono.proj [hunk 1]
```diff
@@ -491,6 +491,7 @@
       <_MonoCMakeBuildCommand Condition="'$(_MonoUseNinja)' != 'true'">$(_MonoCMakeBuildCommand) --parallel $([System.Environment]::ProcessorCount)</_MonoCMakeBuildCommand>
       <_MonoCMakeBuildCommand Condition="'$(TargetsBrowser)' != 'true' and '$(HostOS)' != 'windows'">@(_MonoBuildEnv, ' ') $(_MonoCMakeBuildCommand)</_MonoCMakeBuildCommand>
       <_MonoCMakeBuildCommand Condition="'$(TargetsBrowser)' != 'true' and '$(HostOS)' == 'windows'">call &quot;$(RepositoryEngineeringDir)native\init-vs-env.cmd&quot; $(_CompilerTargetArch) &amp;&amp; cd /D &quot;$(MonoObjDir)&quot; &amp;&amp; @(_MonoBuildEnv, ' ') $(_MonoCMakeBuildCommand)</_MonoCMakeBuildCommand>
+      <_MonoCMakeBuildCommand Condition="'$(TargetsBrowser)' == 'true' and '$(HostOS)' == 'windows'">call &quot;$(RepositoryEngineeringDir)native\init-vs-env.cmd&quot; &amp;&amp; $(_MonoCMakeBuildCommand)</_MonoCMakeBuildCommand>
```
Comment (agocke) [inline src/mono/mono.proj:494]: What's the main difference between this and the above? The addition of `_MonoBuildEnv`?

Comment (hoyosjs) [reply src/mono/mono.proj:494]: yup, $(_CompilerTargetArch) and the build env

Comment (am11) [reply src/mono/mono.proj:494]: Before it was using line 489 on wasm, and it was failing to find cmake on PATH. Now we are prepending it with `eng\native\init-vs-env.cmd &&` so PATH gets updated by VS cmd init script that has cmake bin path.

**Review by agocke — APPROVED** [review]

**Comment by hoyosjs** [comment]
The only case where I am seeing crashes is x86. This seems like a suspension context issue: 
```
00 0785d32c 72997b8e     coreclr!EEPolicy::HandleFatalError+0x72 [D:\a\_work\1\s\src\coreclr\vm\eepolicy.cpp @ 776] 
01 0785d4c4 76f634a1     coreclr!CPFH_VerifyThreadIsInValidState+0x1210d9 [D:\a\_work\1\s\src\coreclr\vm\i386\excepx86.cpp @ 345] 
02 0785d4e8 76f63473     ntdll!ExecuteHandler2+0x26 [d:\w7rtm\minkernel\ntos\rtl\i386\xcptmisc.asm @ 233] 
03 0785d598 76f63414     ntdll!ExecuteHandler+0x24 [d:\w7rtm\minkernel\ntos\rtl\i386\xcptmisc.asm @ 192] 
04 0785d598 76f10133     ntdll!RtlDispatchException+0x127 [d:\w7rtm\minkernel\ntos\rtl\i386\exdsptch.c @ 560] 
05 0785da64 75abf4a5     ntdll!KiUserExceptionDispatcher+0xf [d:\w7rtm\minkernel\ntos\rtl\i386\userdisp.asm @ 503] 
06 0785da64 728cb671     kernel32!SetXStateFeaturesMask+0x8 [d:\w7rtm\base\win32\client\xsave.c @ 296] 
07 0785da8c 72881278     coreclr!Thread::RedirectThreadAtHandledJITCase+0x45 [D:\a\_work\1\s\src\coreclr\vm\threadsuspend.cpp @ 2917] 
08 (Inline) --------     coreclr!Thread::CheckForAndDoRedirect+0xc [D:\a\_work\1\s\src\coreclr\vm\threadsuspend.cpp @ 2998] 
09 (Inline) --------     coreclr!Thread::CheckForAndDoRedirectForGC+0xc [D:\a\_work\1\s\src\coreclr\vm\threadsuspend.cpp @ 3130] 
0a 0785db08 728ccd4c     coreclr!ThreadSuspend::SuspendRuntime+0x3aa [D:\a\_work\1\s\src\coreclr\vm\threadsuspend.cpp @ 3525] 
0b 0785dbb4 728ccbe3     coreclr!ThreadSuspend::SuspendEE+0xf9 [D:\a\_work\1\s\src\coreclr\vm\threadsuspend.cpp @ 5753] 
0c 0785dbdc 72876ba8     coreclr!GCToEEInterface::SuspendEE+0x26 [D:\a\_work\1\s\src\coreclr\vm\gcenv.ee.cpp @ 28] 
0d 0785dbdc 728f9ff9     coreclr!WKS::GCHeap::GarbageCollectGeneration+0xa8 [D:\a\_work\1\s\src\coreclr\gc\gc.cpp @ 45766] 
0e 0785dc00 7281ae50     coreclr!WKS::gc_heap::trigger_gc_for_alloc+0x19 [D:\a\_work\1\s\src\coreclr\gc\gc.cpp @ 17338] 
0f 0785dc34 728f3e68     coreclr!WKS::gc_heap::try_allocate_more_space+0x1e4 [D:\a\_work\1\s\src\coreclr\gc\gc.cpp @ 17484] 
10 0785dc4c 7281a113     coreclr!WKS::gc_heap::allocate_more_space+0x18 [D:\a\_work\1\s\src\coreclr\gc\gc.cpp @ 17948] 
11 (Inline) --------     coreclr!WKS::gc_heap::allocate+0x40 [D:\a\_work\1\s\src\coreclr\gc\gc.cpp @ 17979] 
12 0785dc6c 728183dc     coreclr!WKS::GCHeap::Alloc+0x53 [D:\a\_work\1\s\src\coreclr\gc\gc.cpp @ 44762] 
13 (Inline) --------     coreclr!Alloc+0x108 [D:\a\_work\1\s\src\coreclr\vm\gchelpers.cpp @ 226] 
14 0785dccc 72819dea     coreclr!AllocateObject+0x175 [D:\a\_work\1\s\src\coreclr\vm\gchelpers.cpp @ 979] 
15 0785dd50 06d04ba6     coreclr!JIT_New+0xaa [D:\a\_work\1\s\src\coreclr\vm\jithelpers.cpp @ 2312] 
16 0785dd7c 06d10a7e     Common_Tests!System.Net.Http.Unit.Tests.HPack.HPackDecoderTests.CreateDecoderAndTable()+0x46 [/_/src/libraries/Common/tests/Tests/System/Net/aspnetcore/Http2/HPackDecoderTest.cs @ 99] 
17 0785dddc 06d0fd4f     Common_Tests!System.Net.Http.Unit.Tests.HPack.HPackDecoderTests.TestDecode(Byte[], System.String, System.String, Boolean, Boolean)+0x1e [/_/src/libraries/Common/tests/Tests/System/Net/aspnetcore/Http2/HPackDecoderTest.cs @ 683] 
18 0785ddf8 06d141aa     Common_Tests!System.Net.Http.Unit.Tests.HPack.HPackDecoderTests.TestDecodeWithoutIndexing(Byte[], System.String, System.String)+0x1f [/_/src/libraries/Common/tests/Tests/System/Net/aspnetcore/Http2/HPackDecoderTest.cs @ 678] 
19 0785de10 72909d5f     Common_Tests!System.Net.Http.Unit.Tests.HPack.HPackDecoderTests.DecodesLiteralHeaderFieldNeverIndexed_NewName_HuffmanEncodedNameAndValue()+0x52 [/_/src/libraries/Common/tests/Tests/System/Net/aspnetcore/Http2/HPackDecoderTest.cs @ 355] 
```

Is this known @jkotas @VSadov? Is this safe to merge?

**Comment by jkotas** [comment]
Yes, that is #65890

**Comment by akoeplinger** [comment]
We'll need to backport this to basically all earlier releases right?

**Comment by hoyosjs** [comment]
Yes

**Pull request merged**